### ISSUE TITLE
[exporter/newrelic] instrumentation.provider attribute is missing

### ISSUE THREAD

**Issue by F21** [open-telemetry/opentelemetry-collector-contrib#2621]
**Describe the bug**
According to the opentelemetry spec for new relic, there should be an `instrumentation.provider` attribute:https://github.com/newrelic/newrelic-exporter-specs/blob/master/opentelemetry/OpenTelemetry-Spans.md

This appears to be missing in the exporter and spans sent to new relic are missing this attribute.

**Steps to reproduce**
1. Instrument an app (I used the Go SDK)
2. Point the app to use an opentelemetry collector with the new relic exporter.
3. Filter entities in the explorer with `instrumentation.provider = opentelemetry` and see that the entities do not show up.

**What did you expect to see?**
Spans sent to new relic should have the `instrumentation.provider` attribute set.

**What did you see instead?**
The `instrumentation.provider` attribute was not set.

**What version did you use?**
v0.21.0

**What config did you use?**
```
receivers:
  otlp:
    protocols:
      grpc:
processors:
  batch:
exporters:
  newrelic:
    apikey: NEW_RELIC_KEY_GOES_HERE
    timeout: 30s
service:
  pipelines:
    traces:
      receivers: [otlp]
      processors: [batch]
      exporters: [newrelic]
    metrics:
      receivers: [otlp]
      processors: [batch]
      exporters: [newrelic]
```

**Environment**
OS: Minikube in docker mode

**Comment by gunturaf** [comment]
I came across this issue too, this causes my Traces to only visible on the "Distributed Tracing" page in the One UI APM page, while not being visible as Transactions in the APM.

Firstly I came across this code in the `newrelic/opentelemetry-exporter-go` repo. https://github.com/newrelic/opentelemetry-exporter-go/blob/e563425c51c9e256b329c37e72192baac0396aca/newrelic/internal/transform/span.go#L63

My traces will correctly shown in the Transactions page when I manually add the `instrumentation.provider = opentelemetry` attribute to my Traces. The immediate workaround I did was like this:

```go
serviceAttr := resource.NewWithAttributes(
	semconv.ServiceNameKey.String("something-something"),
	semconv.ServiceNamespaceKey.String("group-of-something"),
	attribute.String("instrumentation.provider", "opentelemetry"), // add this one
)
pusher := otelController.New(
	// other opts...
	otelController.WithResource(serviceAttr),
)
```

I'll volunteer for PR if this issue is confirmed.

cc-ing NR code owners @alanwest @a-feld @jack-berg @nrcventura
